CLI: Generate compodoc command based on user's setup

According to @LeandroCarolino01 the new angular CLI template is not working out of the box in his project https://github.com/storybookjs/storybook/issues/10869#issuecomment-660639116

> [14:50:23] "./tsconfig.base.json" file was not found in the current directory
> npm ERR! code ELIFECYCLE
> npm ERR! errno 1
> npm ERR! testing-sb@0.0.0 docs:json: compodoc -p ./tsconfig.base.json -e json -d .
> npm ERR! Exit status 1

We should generate the `docs:json` command based on the user's setup rather than hard-coding it to the latest `angular-cli`. Help wanted @tooppaaa @kroeder @gaetanmaisse 

@LeandroCarolino01 can you help kick this off by telling us more about your setup?

--- a/cypress/generated/addon-viewport.spec.ts
+++ b/cypress/generated/addon-viewport.spec.ts
@@ -4,7 +4,7 @@ describe('addon-viewport', () => {
   });
 
   it('should have viewport button in the toolbar', () => {
-    cy.navigateToStory('button', 'Text');
+    cy.navigateToStory('example-button', 'Primary');
 
     // Click on viewport button and select small mobile
     cy.get('[title="Change the size of the preview"]').click();

--- a/lib/cli/src/generators/ANGULAR/angular-helpers.ts
+++ b/lib/cli/src/generators/ANGULAR/angular-helpers.ts
@@ -1,5 +1,7 @@
 import * as path from 'path';
 import * as fs from 'fs';
+import { pathExists } from 'fs-extra';
+
 import { readFileAsJson, writeFileAsJson } from '../../helpers';
 
 type TsConfig = {
@@ -51,4 +53,8 @@ export function editStorybookTsConfig(tsconfigPath: string) {
 export function isDefaultProjectSet() {
   const angularJson = readFileAsJson('angular.json', true);
   return angularJson && !!angularJson.defaultProject;
+}
+
+export async function getBaseTsConfigName() {
+  return (await pathExists('./tsconfig.base.json')) ? 'tsconfig.base.json' : 'tsconfig.json';
 }

--- a/lib/cli/src/generators/ANGULAR/index.ts
+++ b/lib/cli/src/generators/ANGULAR/index.ts
@@ -4,6 +4,7 @@ import {
   editStorybookTsConfig,
   getAngularAppTsConfigJson,
   getAngularAppTsConfigPath,
+  getBaseTsConfigName,
 } from './angular-helpers';
 import { writeFileAsJson, copyTemplate } from '../../helpers';
 import { baseGenerator, Generator } from '../baseGenerator';
@@ -40,8 +41,9 @@ const generator: Generator = async (packageManager, npmOptions, options) => {
   editStorybookTsConfig(path.resolve('./.storybook/tsconfig.json'));
 
   // edit scripts to generate docs
+  const tsConfigFile = await getBaseTsConfigName();
   packageManager.addScripts({
-    'docs:json': 'compodoc -p ./tsconfig.base.json -e json -d .',
+    'docs:json': `compodoc -p ./${tsConfigFile} -e json -d .`,
   });
   packageManager.addStorybookCommandInScripts({
     port: 6006,

